src/approaches/nsrt_rl_approach.py

```diff
@@ -61,126 +60,134 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
-                next_rel_param = parent_option.get_rel_option_param_from_state(
-                    next_state,
-                    curr_option.memory,
-                    curr_option.objects,
-                )
-                reward: int = 0
-                if np.allclose(next_rel_param, 0, atol=self._reward_epsilon):
-                    reward += self._pos_reward
-                else:
-                    reward += self._neg_reward
-
-                experience.append(
-                    (state, input_vec, action, reward, next_state))
-
-                # Store experience data in two cases: (1) the next state is a
-                # terminal state for the current option and (2) the current
-                # option did not reach its terminal state but had a sufficient
-                # number of steps to reach it so that it is reasonable to assign
-                # it a reward.
-                terminate = parent_option.effect_based_terminal(
-                    next_state, curr_option.objects)
-                had_sufficient_steps = next_state.allclose(
-                    traj.states[-1]) and (
-                        CFG.max_num_steps_interaction_request -
-                        (idx + 1) > CFG.valid_reward_steps_threshold)
-                if terminate:
-                    option_to_data[parent_option].append(experience)
-                    curr_option_idx += 1
-                    if curr_option_idx < len(plan):
-                        curr_option = plan[curr_option_idx]
-                        parent_option = cast(_LearnedNeuralParameterizedOption,
-                                             curr_option.parent)
-                        experience = []  # Reset for next option in the plan.
-                        if parent_option not in option_to_data:
-                            option_to_data[parent_option] = []
-                    else:
-                        # If we run out of options in the plan, there should be
-                        # an _OptionPlanExhausted exception, and so there is
-                        # nothing more in the trajectory that we have not yet
-                        # assigned to an _Option already. That is, the current
-                        # state in this loop is the last state.
-                        assert next_state.allclose(traj.states[-1])
-                elif had_sufficient_steps:
-                    assert curr_option == plan[-1]
-                    option_to_data[parent_option].append(experience)
+        # For each InteractionResult, compute the experience data for each
+        # _Option we see used in that interaction.
+        for i, result in enumerate(results):
+            option_to_data_from_result = self._get_experience_from_result(
+                i, result)
+            for option, experience in option_to_data_from_result.items():
+                option_to_data.setdefault(option, []).extend(experience)
```

Comment:
this is a little hard to read. maybe use a defaultdict, or just break it up into cases.